src/main/java/com/fasterxml/jackson/core/util/BufferRecyclerPool.java

```diff
@@ -1,17 +1,157 @@
 package com.fasterxml.jackson.core.util;
 
-import com.fasterxml.jackson.core.TokenStreamFactory;
+import java.io.Serializable;
+import java.util.Deque;
+import java.util.concurrent.ConcurrentLinkedDeque;
+import java.util.concurrent.atomic.AtomicReference;
 
 /**
  * Interface for entity that controls creation and possible reuse of {@link BufferRecycler}
  * instances used for recycling of underlying input/output buffers.
  *
  * @since 2.16
  */
-public interface BufferRecyclerPool
-    extends java.io.Serializable
-{
-    public BufferRecycler acquireBufferRecycler(TokenStreamFactory forFactory);
+public interface BufferRecyclerPool extends Serializable {
 
-    public void releaseBufferRecycler(BufferRecycler recycler);
+    BufferRecycler acquireBufferRecycler();
+
+    void releaseBufferRecycler(BufferRecycler recycler);
+
+    /**
+     * Returns the default BufferRecyclerPool implementation which is the thread local based one.
+     */
+    static BufferRecyclerPool defaultRecyclerPool() {
+        return PoolStrategy.THREAD_LOCAL.getPool();
+    }
+
+    enum PoolStrategy {
```

Comment:
I don't understand the point of `enum` here, to be honest. Maybe it becomes more clear after reading these a few times over.

Reply:
I believe that the different pool implementations should be singletons (something you already did in your former commit by using that `static final INSTANCE`) and imo the most idiomatic way to have singletons in Java is using `enum`. Anyway I'm open to change this if you want to suggest a different way of doing this.

Reply:
I think ability to create instances can be useful and important for "real" pools, at least ones with bounded size.
This allows use of one pool per factory, or some other splitting. Basically, it seems useful to not mandate use of only global pools.